packages/next/build/swc/src/styled_jsx/mod.rs

```diff
@@ -304,18 +305,49 @@ impl Fold for StyledJSXTransformer {
         new_items
     }
 
-    fn fold_function(&mut self, func: Function) -> Function {
+    fn fold_binding_ident(&mut self, node: BindingIdent) -> BindingIdent {
+        if self.in_function_params {
+            self.nearest_scope_bindings
+                .insert((node.id.sym.clone(), node.id.span.ctxt));
+        }
+        node
+    }
+
+    fn fold_assign_pat_prop(&mut self, node: AssignPatProp) -> AssignPatProp {
+        if self.in_function_params {
+            self.nearest_scope_bindings
+                .insert((node.key.sym.clone(), node.key.span.ctxt));
+        }
+        node
+    }
+
+    fn fold_function(&mut self, mut func: Function) -> Function {
         let nearest_scope_bindings = self.nearest_scope_bindings.clone();
-        self.nearest_scope_bindings = collect_decls(&func);
-        let func = func.fold_children_with(self);
+        self.nearest_scope_bindings = Default::default();
+        self.in_function_params = true;
```

Comment:
Instead of 

```rust
let nearest_scope_bindings = self.nearest_scope_bindings.clone();
self.nearest_scope_bindings = Default::default();
```

you can do

```rust
let nearest_scope_bindings = take(&mut self.nearest_scope_bindings);
```

This will not clone data, and set `self.nearest_scope_bindings` as `Default::default()`.

Reply:
Oh nice!